pkgs/apps/meep/default.nix

```diff
@@ -71,27 +71,41 @@ buildPythonPackage rec {
 
   checkInputs = [ openssh ];
   doCheck = true;
-  checkPhase = ''
-    export PYTHONPATH="$out/lib/${python.libPrefix}/site-packages:$PYTHONPATH"
-    python3 << EOF
-    import meep as mp
-    cell = mp.Vector3(16,8,0)
-    geometry = [mp.Block(mp.Vector3(mp.inf,1,mp.inf),
-                     center=mp.Vector3(),
-                     material=mp.Medium(epsilon=12))]
-    sources = [mp.Source(mp.ContinuousSource(frequency=0.15),
-                     component=mp.Ez,
-                     center=mp.Vector3(-7,0))]
-    pml_layers = [mp.PML(1.0)]
-    resolution = 10
-    sim = mp.Simulation(cell_size=cell,
-                    boundary_layers=pml_layers,
-                    geometry=geometry,
-                    sources=sources,
-                    resolution=resolution)
-    sim.run(until=200)
-    EOF
-  '';
+  checkPhase =
+    let testInput = writeTextFile {
```

Comment:
I would rather keep the
```
cat  > test.py << EOF
...
EOF
```
construction here. That should be faster than building the text file.

Reply:
OK, can change it again. I've removed the original version as only one of the launched interpreters got the input. But of course we can use cat as you suggest.